ecommerce-ui/src/state/product/reducer.js

```diff
@@ -0,0 +1,38 @@
+import {
+  FETCH_PRODUCTS,
+  FETCH_PRODUCTS_SUCCESS,
+  FETCH_PRODUCTS_FAILURE,
+} from '../actionTypes';
+
+const initialState = {
+  isLoading: false,
+  products: [],
+  error: null,
+};
+
+export default (state = initialState, action) => {
```

Comment:
Nice job making a reducer. For larger apps, this makes state a lot more manageable.